### PR TITLE
support for old bugged nil return functions as well as fixed ones

### PR DESCRIPTION
**Pull request by bartkrak**
solves: https://github.com/membraneframework/membrane_core/issues/758

Not so pretty but it enough for a temporary fix that will be deleted someday

### CODE REVIEW

File: lib/unifex/code_generators/nif.ex [hunk 1]
```diff
@@ -456,4 +467,58 @@ defmodule Unifex.CodeGenerators.NIF do
...
+        ~g<UNIFEX_TERM #{[name, :result, ""] |> Enum.join("_")}(#{args_declarations})>} {
+          return #{BaseType.generate_arg_serialize(:atom, :"\"#{nil}\"", NIF, ctx)};
+        }
+        """
+      else
+        ~g<>
+      end
+    end)
+    |> Enum.filter(&(&1 != ""))
+    |> Enum.map_join("\n", & &1)
```
Comment (FelonEkonom) [inline lib/unifex/code_generators/nif.ex:522]: You can use just `Enum.join/2` here and in similar places

File: lib/unifex/code_generators/nif.ex [hunk 2]
```diff
@@ -456,4 +467,58 @@ defmodule Unifex.CodeGenerators.NIF do
...
+
+        ~g<UNIFEX_TERM #{[name, :result, ""] |> Enum.join("_")}(#{args_declarations})>} {
+          return #{BaseType.generate_arg_serialize(:atom, :"\"#{nil}\"", NIF, ctx)};
+        }
+        """
+      else
+        ~g<>
+      end
+    end)
+    |> Enum.filter(&(&1 != ""))
```
Comment (FelonEkonom) [inline lib/unifex/code_generators/nif.ex:521]: Instead of combining `Enum.map/2` and `Enum.filter/2` you can just use `Enum.flat_map/2` here in similar places. It will simplify your code

File: lib/unifex/code_generators/nif.ex [hunk 3]
```diff
@@ -456,4 +467,58 @@ defmodule Unifex.CodeGenerators.NIF do
...
+  # This function generates bugged version of functions declarations returning nil and should be removed in v2.0.0
+  defp generate_functions_declarations_bugged(config, ctx) do
+    config
+    |> Enum.map(fn {name, result} ->
+      {_result, meta} = generate_serialization(result, ctx)
+      args = meta |> Keyword.get_values(:arg)
+
+      args_declarations =
+        [~g<UnifexEnv* env> | generate_args_declarations(args, :const_unless_ptr_on_ptr, ctx)]
+        |> Enum.join(", ")
```
Comment (FelonEkonom) [inline lib/unifex/code_generators/nif.ex:480]: Move it inside `if` statement below

File: lib/unifex/code_generators/nif.ex [hunk 4]
```diff
@@ -456,4 +467,58 @@ defmodule Unifex.CodeGenerators.NIF do
...
+        meta
+        |> Keyword.get_values(:label)
+
+      if Enum.member?(labels, "nil") do
+        ~g<UNIFEX_TERM #{[name, :result, ""] |> Enum.join("_")}(#{args_declarations})>
+      else
+        ~g<>
+      end
+    end)
+    |> List.flatten()
```
Comment (FelonEkonom) [inline lib/unifex/code_generators/nif.ex:492]: I guess this `List.flatten` is unnecessary

**Review by FelonEkonom — CHANGES_REQUESTED** [review]
I don't see bugged functions generated in fixtures. Add test checking, if they are generated correctly (remember about `(nil :: label)` and `{nil :: label, reason :: integer}` as result specs)

File: lib/unifex/code_generators/nif.ex (hunk 1 above)
Comment (FelonEkonom) [reply lib/unifex/code_generators/nif.ex:523]: Still unchanged

File: lib/unifex/code_generators/nif.ex (hunk 2 above)
Comment (FelonEkonom) [reply lib/unifex/code_generators/nif.ex]: Unchanged

**Review by FelonEkonom — CHANGES_REQUESTED** [review]
Please, implement or comment unsolved suggestions from previous CR round

**Review by FelonEkonom — APPROVED** [review]
